src/main/java/duke/command/Duke.java

```diff
@@ -0,0 +1,231 @@
+                            + SPACING + NO_ARGUMENT_1 + "delete" + NO_ARGUMENT_2 + "\n"
+                            + LINE_SEPARATOR;
+                    System.out.println(error);
+                }
+                break;
+            default:
+                String response = LINE_SEPARATOR
+                        + SPACING + CANNOT_IDENTIFY + "\n"
+                        + LINE_SEPARATOR;
+                System.out.println(response);
+            }
+        }
+
+        String Bye = LINE_SEPARATOR
+                + SPACING + "Bye. Hope to see you again soon!\n"
+                + LINE_SEPARATOR;
+        System.out.println(Bye);
+    }
+
+
+    public static void list() {
+        String listOutput = LINE_SEPARATOR;
+        for (int i = 0; i < taskList.size(); i++) {
+            listOutput += SPACING + (i + 1) + ". "
+                    + taskList.get(i).toString() + "\n";
+        }
+        listOutput += LINE_SEPARATOR;
+        System.out.println(listOutput);
+
+    }
+    public static void readData() {
+        String filePath = "data/data.txt";
+        try {
+            File f = new File(filePath);
+            Scanner s = new Scanner(f);
+            while (s.hasNext()) {
+                String currentLine = s.nextLine();
+                String[] currentData = currentLine.split(" \\* ");
+                switch (currentData[0]) {
+                case "T":
+                    taskList.add(new ToDo(currentData[2]));
+                    if (currentData[1].equals("1")) {
+                        taskList.get(taskList.size() - 1).markAsDone();
+                    }
+                    break;
+                case "D":
+                    taskList.add(new Deadline(currentData[2], currentData[3]));
+                    if (currentData[1].equals("1")) {
+                        taskList.get(taskList.size() - 1).markAsDone();
+                    }
+                    break;
+                case "E":
+                    taskList.add(new Event(currentData[2], currentData[3]));
+                    if (currentData[1].equals("1")) {
+                        taskList.get(taskList.size() - 1).markAsDone();
+                    }
+                    break;
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("Something is wrong...");
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("File corrupted!");
+        }
+    }
+
+    public static void writeData () {
+        String ddlIndicator = "(by:";
+        int statusIdx = 4;
+        int contentStartIdx = 7;
+        String filePath = "data/data.txt";
+        String updatedData = "";
+        int indicatorLength = 5;
+        for (int i = 0; i < taskList.size(); i++) {
+            String currentTask = taskList.get(i).toString();
+            String type = currentTask.substring(1,2);
+            String status = (currentTask.charAt(statusIdx) == 'X') ? "1" : "0";
```

Comment:
Avoid complicated expressions. You may want to extract this method out, or use a more long winded way to represent it instead of shortcut, it is not clear what it does.